sam_view.c

```diff
@@ -817,6 +825,37 @@ static int getLength(char **s)
     return n;
 }
 
+void copy_tag(const char *tag, const bam1_t *rec, kstring_t *linebuf)
+{
+    uint8_t *s = bam_aux_get(rec, tag);
+    if (s) {
+        char aux_type = *s;
+        switch (aux_type) {
+            case 'C':
+            case 'I':
+            case 'S': aux_type = 'I'; break;
+            case 'c':
+            case 's': aux_type = 'i'; break;
+            case 'd': aux_type = 'f'; break;
+        }
+
+        kputc('\t', linebuf);
+        kputsn(tag, 2, linebuf);
+        kputc(':', linebuf);
+        kputc(aux_type=='I'? 'i': aux_type, linebuf);
+        kputc(':', linebuf);
+        switch (aux_type) {
+            case 'H':
+            case 'Z': kputs(bam_aux2Z(s), linebuf); break;
+            case 'i': kputw(bam_aux2i(s), linebuf); break;
+            case 'I': kputuw(bam_aux2i(s), linebuf); break;
+            case 'A': kputc(bam_aux2A(s), linebuf); break;
+            case 'f': kputd(bam_aux2f(s), linebuf); break;
+            default:  kputs("*** Unknown Type ***", linebuf); break;
```

Comment:
Ah yes I see what it's trying to do, however at the moment if you run:

```
./samtools fastq -t  ../htslib/test/auxf#values.sam
```

you get this output:

```
@Fred	RG:Z:ID
CTGAGCTAGC
+
**********
@Jim	BC:B:*** Unknown Type ***
TTTTTTTTTT
+
""""""""""
```

I see now it's pure fluke my exhaustive htslib tag testing file happens to be using BC (amongst many others) which is an official tag and so it's being output by default.

I'd still prefer a stderr message or similar (emitted once rather than spammed), or even a hard failure.   If we've explicitly asked for tag XX and we can't emit it then that is probably a failure condition.  Inlining the complain to the fastq output is just going to byte us later on.